backupccl: only backup public indexes

pbardea opened the pull request:
This commit ensures that backup only backs up PUBLIC indexes. This means
that it will not back up ADDING indexes.

This is because AddSSTable requests may write in the past, so
incremental backups may miss writes when performing a time-based scan
from the previous incremental backup.

Imforms https://github.com/cockroachdb/cockroach/issues/62564.

Test test previously failed with:
```
--- FAIL: TestBackupOnlyPublicSpans (1.92s)
    backup_test.go:7802: 
                Error Trace:    backup_test.go:7802
                Error:          Received unexpected error:
                                expected incremental backup to not contain any data, found spans [/Table/53/{2-3}]
```

Release note (bug fix): Fixes a bug where index backfill data may have
been missed by BACKUP in incremental backups.

pbardea committed:
backupccl: only backup public indexes

This commit ensures that backup only backs up PUBLIC indexes. This means
that it will not back up ADDING indexes.

This is because AddSSTable requests may write in the past, so
incremental backups may miss writes when performing a time-based scan
from the previous incremental backup.

Release note (bug fix): Fixes a bug where index backfill data may have
been missed by BACKUP in incremental backups.

pbardea committed:
backupccl: test that only public spans are backed up

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62572)
<!-- Reviewable:end -->

pbardea commented:
I added a bit more testing and commentary, this is RFAL.

pbardea committed:
backupccl: clean up mid schema change tests

The suite of tests that test restorations of backups that were taken
mid-schema change is updated to not require that the schema changes
succeed. This is in preparation for the change later in this PR that
stops retoring adding indexes. This could cause restore schema change
jobs to fail as their progress will no longer be in-sync with the
backfilled data.

Release note: None

pbardea commented:
TFTR.
bors r+
